### PR TITLE
README.md: fix quic/ subdir listing

### PR DESCRIPTION
**Pull request by jsquyres**
Trivial update to README.md: Fix a typo (sample->sample) and put the subdirs in alphabetical order.

Signed-off-by: Jeff Squyres <jeff@squyres.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,17 +21,17 @@
 ## Source Layout
 - `quic/api`:         Defines API that applications can use to interact with the QUIC transport  layer.
```
Comment (yangchi) [inline README.md]: Do use a favor and remove the extra space between "transport" and "layer"?

File: README.md [hunk 2]
```diff
@@ -21,17 +21,17 @@
...
 - `quic/client`:      Client transport implementation
-- `quic/server`:      Server transport implementation
 - `quic/codec`:       Read and write codec implementation for the protocol
 - `quic/common`:      Implementation of common utility functions
 - `quic/congestion_control`: Implementation of different congestion control algorithms such  as Cubic and Copa
 - `quic/flowcontrol`: Implementations of flowcontrol functions
```
Comment (yangchi) [inline README.md]: nit: "flow control". Two words.

File: README.md (hunk 1 above)
Comment (jsquyres) [reply README.md]: Done!  Also did the same on the `congestion_control` line.

**Comment by jsquyres** [comment]
@yangchi Fixed!

**Pull request closed without merging**